sale_lead_time_profile/models/lead_time_profile.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2025 Quartile
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class LeadTimeProfile(models.Model):
+    _name = "lead.time.profile"
+
+    warehouse_id = fields.Many2one("stock.warehouse", required=True)
+    partner_id = fields.Many2one(
+        "res.partner", string="Delivery Address", domain="[('type', '=', 'delivery')]"
+    )
+    state_id = fields.Many2one(
+        related="partner_id.state_id",
+        readonly=False,
+        store=True,
```

Comment:
Can we try another implementation. See my suggestion on `_compute_delivery_lead_time()`.
```suggestion
        "res.country.state",
```